homeassistant/components/rainmachine/switch.py

```diff
@@ -191,38 +222,31 @@ def __init__(
         self,
         coordinator: DataUpdateCoordinator,
         controller: Controller,
-        uid: int,
-        name: str,
         entry: ConfigEntry,
+        description: RainMachineSwitchDescription,
     ) -> None:
         """Initialize a generic RainMachine switch."""
-        super().__init__(coordinator, controller, type(self).__name__)
+        super().__init__(coordinator, controller, description)
+        self.entity_description: RainMachineSwitchDescription
```

Comment:
Type this as the class attribute instead

```py
class RainMachineSwitch(RainMachineEntity, SwitchEntity):
    """A class to represent a generic RainMachine switch."""

    _attr_icon = DEFAULT_ICON
    entity_description: RainMachineSwitchDescription
```

Reply:
@bachya I pushed a small commit with these changes so the PR can be ready for the beta cut